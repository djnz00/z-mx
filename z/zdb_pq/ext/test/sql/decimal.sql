SELECT - '42.01'::zdecimal;
SELECT - '0'::zdecimal;
SELECT - 'nan'::zdecimal;
SELECT '55'::zdecimal;
SELECT '-55'::zdecimal;
SELECT ''::zdecimal;
SELECT 'x'::zdecimal;
SELECT '55 x'::zdecimal;
SELECT '127'::zdecimal;
SELECT '200'::zdecimal;
SELECT '-128'::zdecimal;
SELECT '-200'::zdecimal;

CREATE TABLE test_zdecimal (a zdecimal);
INSERT INTO test_zdecimal VALUES (-2), (-1), (0), (1), (2);
SELECT a FROM test_zdecimal;
SELECT a FROM test_zdecimal ORDER BY a;
DROP TABLE test_zdecimal;

SELECT '2'::zdecimal < 5;
SELECT 2 < '5'::zdecimal;
SELECT '5'::zdecimal < 2;
SELECT 5 < '2'::zdecimal;
SELECT '2'::zdecimal <= 5;
SELECT 2 <= '5'::zdecimal;
SELECT '5'::zdecimal <= 2;
SELECT 5 <= '2'::zdecimal;
SELECT '2'::zdecimal = 5;
SELECT 2 = '5'::zdecimal;
SELECT '5'::zdecimal = 2;
SELECT 5 = '2'::zdecimal;
SELECT '2'::zdecimal <> 5;
SELECT 2 <> '5'::zdecimal;
SELECT '5'::zdecimal <> 2;
SELECT 5 <> '2'::zdecimal;
SELECT '2'::zdecimal >= 5;
SELECT 2 >= '5'::zdecimal;
SELECT '5'::zdecimal >= 2;
SELECT 5 >= '2'::zdecimal;
SELECT '2'::zdecimal > 5;
SELECT 2 > '5'::zdecimal;
SELECT '5'::zdecimal > 2;
SELECT 5 > '2'::zdecimal;
SELECT '2'::zdecimal + 5;
SELECT 2 + '5'::zdecimal;
SELECT '5'::zdecimal + 2;
SELECT 5 + '2'::zdecimal;
SELECT '2'::zdecimal - 5;
SELECT 2 - '5'::zdecimal;
SELECT '5'::zdecimal - 2;
SELECT 5 - '2'::zdecimal;
SELECT '2'::zdecimal * 5;
SELECT 2 * '5'::zdecimal;
SELECT '5'::zdecimal * 2;
SELECT 5 * '2'::zdecimal;
SELECT '2'::zdecimal / 5;
SELECT 2 / '5'::zdecimal;
SELECT '5'::zdecimal / 2;
SELECT 5 / '2'::zdecimal;
SELECT '1'::zdecimal < '1'::zdecimal;
SELECT '5'::zdecimal < '2'::zdecimal;
SELECT '3'::zdecimal < '4'::zdecimal;
SELECT '1'::zdecimal <= '1'::zdecimal;
SELECT '5'::zdecimal <= '2'::zdecimal;
SELECT '3'::zdecimal <= '4'::zdecimal;
SELECT '1'::zdecimal = '1'::zdecimal;
SELECT '5'::zdecimal = '2'::zdecimal;
SELECT '3'::zdecimal = '4'::zdecimal;
SELECT '1'::zdecimal <> '1'::zdecimal;
SELECT '5'::zdecimal <> '2'::zdecimal;
SELECT '3'::zdecimal <> '4'::zdecimal;
SELECT '1'::zdecimal >= '1'::zdecimal;
SELECT '5'::zdecimal >= '2'::zdecimal;
SELECT '3'::zdecimal >= '4'::zdecimal;
SELECT '1'::zdecimal > '1'::zdecimal;
SELECT '5'::zdecimal > '2'::zdecimal;
SELECT '3'::zdecimal > '4'::zdecimal;

SELECT zdecimalcmp('1'::zdecimal, '1'::zdecimal);
SELECT zdecimalcmp('5'::zdecimal, '2'::zdecimal);
SELECT zdecimalcmp('3'::zdecimal, '4'::zdecimal);
SELECT pg_typeof('1'::zdecimal + '1'::zdecimal);
SELECT '1'::zdecimal + '1'::zdecimal;
SELECT '3'::zdecimal + '4'::zdecimal;
SELECT '5'::zdecimal + '2'::zdecimal;
SELECT '-3'::zdecimal + '4'::zdecimal;
SELECT '-5'::zdecimal + '2'::zdecimal;
SELECT '3'::zdecimal + '-4'::zdecimal;
SELECT '5'::zdecimal + '-2'::zdecimal;
SELECT '127'::zdecimal + '127'::zdecimal;
SELECT pg_typeof('1'::zdecimal - '1'::zdecimal);
SELECT '1'::zdecimal - '1'::zdecimal;
SELECT '3'::zdecimal - '4'::zdecimal;
SELECT '5'::zdecimal - '2'::zdecimal;
SELECT '-3'::zdecimal - '4'::zdecimal;
SELECT '-5'::zdecimal - '2'::zdecimal;
SELECT '3'::zdecimal - '-4'::zdecimal;
SELECT '5'::zdecimal - '-2'::zdecimal;
SELECT pg_typeof('1'::zdecimal * '1'::zdecimal);
SELECT '1'::zdecimal * '1'::zdecimal;
SELECT '3'::zdecimal * '4'::zdecimal;
SELECT '5'::zdecimal * '2'::zdecimal;
SELECT '-3'::zdecimal * '4'::zdecimal;
SELECT '-5'::zdecimal * '2'::zdecimal;
SELECT '3'::zdecimal * '-4'::zdecimal;
SELECT '5'::zdecimal * '-2'::zdecimal;
SELECT '127'::zdecimal * '127'::zdecimal;
SELECT pg_typeof('1'::zdecimal / '1'::zdecimal);
SELECT '1'::zdecimal / '1'::zdecimal;
SELECT '3'::zdecimal / '4'::zdecimal;
SELECT '5'::zdecimal / '2'::zdecimal;
SELECT '-3'::zdecimal / '4'::zdecimal;
SELECT '-5'::zdecimal / '2'::zdecimal;
SELECT '3'::zdecimal / '-4'::zdecimal;
SELECT '5'::zdecimal / '-2'::zdecimal;
SELECT '5'::zdecimal / '0'::zdecimal;

SELECT CAST('42.01'::zdecimal AS int8);
SELECT CAST('-42.01'::zdecimal AS int8);

SELECT CAST('42.01'::zdecimal AS double precision);
SELECT CAST('-42.01'::zdecimal AS double precision);

SELECT CAST('42.01'::zdecimal AS numeric);
SELECT CAST('-42.01'::zdecimal AS numeric);

SELECT zdecimal_hash(42.01);

SELECT zdecimal_hash(42.01::zdecimal);
