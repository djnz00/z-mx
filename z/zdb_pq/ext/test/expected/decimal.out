SELECT - '5'::int1;
 ?column? 
----------
 -5
(1 row)

SELECT - '127'::int1;
 ?column? 
----------
 -127
(1 row)

SELECT - '-128'::int1;
ERROR:  integer out of range
-- reference
SELECT hashint4(55);
  hashint4  
------------
 -305105437
(1 row)

SELECT hashoid(55);
  hashoid   
------------
 -305105437
(1 row)

SELECT hashint1(55::int1);
  hashint1  
------------
 -305105437
(1 row)

SELECT hashuint1(55::uint1);
 hashuint1  
------------
 -305105437
(1 row)

SELECT hashuint2(55::uint2);
 hashuint2  
------------
 -305105437
(1 row)

SELECT hashuint4(55::uint4);
 hashuint4  
------------
 -305105437
(1 row)

SELECT hashuint8(55::uint8);
 hashuint8  
------------
 -305105437
(1 row)

